csharp-basic-topics/TypeCheckingAndCastingInCSharp/TypeCheckingAndCastingInCSharp.Tests/TypeConversionTests.cs

```diff
@@ -0,0 +1,159 @@
+
+            Assert.Throws<InvalidCastException>(() =>
+            {
+                Animal animal = new Monkey();
+                Horse horse = (Horse)animal;    // compiles but throws InvalidCastException
+            });
+        }
+
+        [Fact]
+        public void WhenUsingAsOperator_ThenExplicitConversionIsDoneOrReturnsNull()
+        {
+            // act
+            Animal animal = new Horse();
+
+            Horse? horse = animal as Horse; // explicit conversion
+            Monkey? monkey = animal as Monkey;  // invalid conversion, monkey contains null
+
+            // assert
+            Assert.Same(animal, horse);
+            Assert.Null(monkey);
+        }
+
+        [Fact]
+        public void WhenConvertingFromString_ThenUseParseMethod()
+        {
+            // act
+            int number = int.Parse("10");   // convert string to integer
+            bool boolean = bool.Parse("True");  // convert string to boolean
+
+            // assert
+            Assert.Equal(10, number);
+            Assert.True(boolean);
+        }
+
+        [Fact]
+        public void WhenConvertingFromString_ThenUseTryParseMethod()
+        {
+            // act
+            var succeed = int.TryParse("10", out var number);
+            // succeed == true
+            // number == 10
+
+            // assert
+            Assert.Equal(10, number);
+            Assert.True(succeed);
+        }
+
+        [Fact]
+        public void WhenConvertingBasicTypes_ThenUseConvertStaticMethods()
+        {
+            // arrange
+            decimal fraction = 12.5M;
+
+            // act
+            int integer = Convert.ToInt16(fraction);
+            string str = Convert.ToString(fraction);
+            bool boolean = Convert.ToBoolean("true");
+
+            // assert
+            Assert.Equal(12, integer);
+            Assert.Equal("12.5", str);
+            Assert.True(boolean);
+        }
+
+        [Fact]
+        public void WhenConvertingBasicTypesToBytes_ThenUseBitConverter()
+        {
+            // arrange
+            var sw = new StringWriter();
+            Console.SetOut(sw);
+
+            int integer = 1024;
+            float fraction = 12.5F;
+
+            // act
+            var integerAsBytes = BitConverter.GetBytes(integer);
+            var floatAsBytes = BitConverter.GetBytes(fraction);
+
+            var format = "{0,12}{1,15}";
+            Console.WriteLine(format, "Base Type", "Bytes");
+            Console.WriteLine(format, "=========", "=====");
+            Console.WriteLine(format, integer, BitConverter.ToString(integerAsBytes));
+            Console.WriteLine(format, fraction, BitConverter.ToString(floatAsBytes));
+
+            // Assert
+            Assert.Equal("   Base Type          Bytes" + Environment.NewLine + "   =========          =====" + Environment.NewLine + "        1024    00-04-00-00" + Environment.NewLine + "        12.5    00-00-48-41" + Environment.NewLine, sw.ToString());
```

Comment:
Please no SetOut method for tests. Just had a discussion with Marinko the other day. It's unreliable and hard to create.
This is easy to test without it:
![image](https://user-images.githubusercontent.com/29179238/182671230-4299085f-97a8-4ed3-88ba-d50ba3d072df.png)

Reply:
fixed, following your example